### DISCUSSION TITLE
Routing over SSTP is broken or what ?

### DISCUSSION THREAD

**Discussion by AMajor-C** [SoftEtherVPN/SoftEtherVPN#1985] [Q&A]
I have SE server latest version and 2 routers with sstp mikrotik and keenetic.
On both routers i cannot setup routing over sstp.
It seems it caused by "Local" and "Remote" adresses.
If we using dhcp on virtual-nat we receive local adress from pool and thats good but remote address is always virtual-nat ip. **But virtual-nat cannot route traffic!** 
On both routers we can provide local and remote address manually but if we define other address the tunnel just dont up. if we manually put route to network over interface it will not work also.
Both routers can access the network created by virtual-nat directly. But routing form networks behind doesnt work.
It seems in needs to combine L3 Switch and dhcp functionality or add routing fature to virtual-nat ?
Or i do something wrong ?
@chipitsine

**Comment by shakibamoshiri** [comment]
I have connected MT to SE dev over SSTP with no issue 

```bash
[ client (sstp) ] => [ MT (hop-1) ] => [ SE dev (hop-2) ] => Internet 
```

your condition is a bit unclear to me

**Reply by AMajor-C** [reply]
For dumb connetion to internet of course you have no issue.
According your shema:
[a host subnet 1] -> [MT subnet 1] -> [SE vnat] -> internet  - works 
[a host subnet 1] -> [MT subnet 1] -> [SE vnat] -> [Keen subnet 2 host]  - doesnt work
[a host subnet 2] -> [Keen subnet 2] -> [SE vnat] -> [MT subnet 1 host]  - doesnt work
Because keenectic doenst work correctly with manual added local-remote
[a host subnet 1] -> [MT subnet 1] -> [SE vswitch] -> [Keen subnet 2 host]  -  works
[a host subnet 2] -> [Keen subnet 2] -> [SE vswitch] -> [MT subnet 1 host]  - doesnt work

**Comment by shakibamoshiri** [comment]
I got it almost , SE virtual NAT only routes what ever the traffic is to its host (running on)
actually it is doing  DNAT so the users can access the host net.

if you want to send/forward your request/traffic from SE to further application , you better to use
- virtual Switch 
- manual bridge (on linux)

and if you cannot use Linux , so the only choice remains will be  vSwitch

I have tested SE vSwitch, it has worked fine, but not your scenario

**Reply by Evengard** [reply]
SSTP is based on PPP, and PPP is a layer 3 point to point protocol, it needs a remote endpoint to just work. And this is that endpoint which will actually do the complex routing if needed.
The only thing a client routing (through manual config or DHCP) can do is decide what is to be routed INSIDE the tunnel or OUTSIDE of it.
